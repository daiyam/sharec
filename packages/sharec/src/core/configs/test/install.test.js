const { fixtures } = require('testUtils')
const { vol } = require('memfs')
const { installConfig } = require('../install')

describe('processors > configs >', () => {
  const eslintBaseFxt = fixtures('eslint/json/01-base', 'json')
  const eslintBaseFxtYaml = fixtures('eslint/yaml/01-base')

  beforeEach(() => {
    vol.reset()
  })

  describe('installConfig', () => {
    it('should merge JSON configs', async () => {
      expect.assertions(1)

      const dir = {
        '/target/.eslintrc': JSON.stringify(eslintBaseFxt.current),
      }

      vol.fromJSON(dir, '/')

      await installConfig({
        targetPath: '/target',
        configPath: '.eslintrc',
        configSource: JSON.stringify(eslintBaseFxt.upcoming),
      })

      expect(JSON.parse(vol.readFileSync('/target/.eslintrc', 'utf8'))).toEqual(
        eslintBaseFxt.result,
      )
    })

    it('should merge YAML configs', async () => {
      expect.assertions(1)

      const dir = {
        '/target/.eslintrc.yaml': eslintBaseFxtYaml.current,
      }
      vol.fromJSON(dir, '/')

      await installConfig({
        targetPath: '/target',
        configPath: '.eslintrc.yaml',
        configSource: eslintBaseFxtYaml.upcoming,
      })

      expect(vol.readFileSync('/target/.eslintrc.yaml', 'utf8')).toEqual(
        eslintBaseFxtYaml.result,
      )
    })

    it('should copy all non-mergeable configs', async () => {
      expect.assertions(1)

      const dir = {}

      vol.fromJSON(dir, '/')

      await installConfig({
        targetPath: '/target',
        configPath: '.editorconfig',
        configSource: 'bar',
      })

      const res = await vol.readFileSync('/target/.editorconfig', 'utf8')

      expect(res).toEqual('bar')
    })

    it('should correctly process configs with mixed file structure', async () => {
      expect.assertions(1)

      const dir = {}
      vol.fromJSON(dir, '/')

      await installConfig({
        targetPath: '/target',
        configPath: 'foo/bar/.editorconfig',
        configSource: 'bar',
      })

      expect(vol.readFileSync('/target/foo/bar/.editorconfig', 'utf8')).toEqual(
        'bar',
      )
    })
  })
})
