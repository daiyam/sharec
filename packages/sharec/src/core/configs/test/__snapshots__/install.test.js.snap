// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`processors > configs > installConfig should merge JSON configs 1`] = `
Object {
  "env": Object {
    "browser": true,
    "es6": true,
    "jest": true,
    "node": true,
  },
  "extends": "airbnb",
  "globals": Object {
    "bar": true,
    "foo": true,
  },
  "parserOptions": Object {
    "ecmaFeatures": Object {
      "spread": true,
    },
    "ecmaVersion": 7,
    "parser": "babel-eslint",
    "sourceType": "module",
  },
  "rules": Object {
    "max-len": Array [
      "warn",
      120,
      4,
      Object {
        "ignoreComments": true,
        "ignoreStrings": false,
        "ignoreUrls": false,
      },
    ],
    "no-console": 0,
    "space-before-function-paren": 1,
  },
}
`;

exports[`processors > configs > installConfig should merge YAML configs 1`] = `
"env:
  browser: true
  es6: true
  node: true
  jest: true
globals:
  foo: true
  bar: true
extends: airbnb
parserOptions:
  parser: babel-eslint
  ecmaVersion: 7
  ecmaFeatures:
    spread: true
  sourceType: module
rules:
  space-before-function-paren: 1
  no-console: 0
  max-len:
    - warn
    - 120
    - 4
    - ignoreUrls: false
      ignoreComments: true
      ignoreStrings: false
"
`;
